### PR TITLE
[JUJU-1081] Fixes expose ipv6 issue in Openstack.

### PR DESCRIPTION
**Pull request by tlm**
It's possible that when exposing an application in Juju that is deployed
across an Openstack provider the provider will only expost v4 cidr's and
not v6 cidr's. This happens because the cidr list to expose to is empty
so the provider defaults it off to v4 only.

This change doesn't reflect what happens elsewhere. When looking at the
facade for expose you can see that v6 cidr ranges are added to an expose
when none are provided.

This change now adds both v4 and v6 cidr ranges should none be provided
by the user.

## Checklist

 - [x] Requires a [pylibjuju](https://github.com/juju/python-libjuju) change
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR
 - [x] Added or updated [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) related to packages changed
 - [x] Comments answer the question of why design decisions were made

## QA steps

The best QA steps come from the bug report and also the new regression tests.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1666537

### CODE REVIEW

**Review by wallyworld — APPROVED** [review]

File: provider/openstack/provider.go [hunk 1]
```diff
@@ -2080,7 +2080,7 @@ func rulesToRuleInfo(groupId string, rules firewall.IngressRules) []neutron.Rule
 		}
 		sourceCIDRs := r.SourceCIDRs.Values()
 		if len(sourceCIDRs) == 0 {
-			sourceCIDRs = append(sourceCIDRs, firewall.AllNetworksIPV4CIDR)
+			sourceCIDRs = append(sourceCIDRs, firewall.AllNetworksIPV4CIDR, firewall.AllNetworksIPV6CIDR)
```
Comment (wallyworld) [inline provider/openstack/provider.go:2083]: assume this will be ok on clouds that don't support v6?

Comment (tlm) [reply provider/openstack/provider.go:2083]: yeah Openstack doesn't care.

**Pull request merged**